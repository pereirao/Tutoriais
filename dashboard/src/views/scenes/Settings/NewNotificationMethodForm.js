// @flow

import React from 'react';
import {connect} from 'react-redux';
import classNames from 'classnames/bind';
import _ from 'lodash';
import style_sheet from './Settings.module.css';
import NewFormError from './NewFormError';
import I18n from '../../../localization';
import {notification_method_selectors, notification_method_actions} from '../../../business/notification_method';
import {MBForm, MBFormSubmitButton} from '../../elements/MBForm';
import {
  NameField,
  NotificationTypeField
} from '../../shared/form_fields';

const cx = classNames.bind(style_sheet);

type NewNotificationMethodFormProps = {
  createNotificationMethod: (Object) => void,
  close: () => void,
  resetForm: () => void,
  create_notification_method_errors: Array<string>,
  is_loading: boolean,
  successfully_created: boolean
};
export const NewNotificationMethodForm = ({
                                            createNotificationMethod,
                                            close,
                                            resetForm,
                                            create_notification_method_errors,
                                            is_loading,
                                            successfully_created
                                          }: NewNotificationMethodFormProps) => {
  if (successfully_created) {
    close();
  }
  return (
    <MBForm
      onSubmit={(values) => {
        resetForm();
        createNotificationMethod(values);
      }}
      className={cx('newForm', {shake: !_.isEmpty(create_notification_method_errors)})}>
      <NotificationTypeField/>
      <div className={style_sheet.fieldLabel}>{I18n.t('form.label.value')}</div>
      <NameField name="value" placeholder={I18n.t('form.placeholder.notification_method_value')}/>
      <div className={style_sheet.fieldLabel}>{I18n.t('form.label.label')}</div>
      <NameField name="label" placeholder={I18n.t('form.placeholder.notification_method_label')} is_required={false}/>
      <MBFormSubmitButton is_loading={is_loading} text={I18n.t('ui.modal.new_notification_method.submit')}/>
      <NewFormError error_messages={create_notification_method_errors}/>
    </MBForm>
  );
};

const NewNotificationMethodFormSTP = state => ({
  create_notification_method_errors: notification_method_selectors.createNotificationMethodErrors(state),
  is_loading: notification_method_selectors.isCreating(state),
  successfully_created: notification_method_selectors.successfullyCreated(state)
});
const NewNotificationMethodFormDTP = {
  createNotificationMethod: notification_method_actions.createNotificationMethod,
  resetForm: notification_method_actions.resetNewNotificationMethodForm
};
const NewNotificationMethodFormContainer = connect(NewNotificationMethodFormSTP, NewNotificationMethodFormDTP)(NewNotificationMethodForm);

export default NewNotificationMethodFormContainer;
