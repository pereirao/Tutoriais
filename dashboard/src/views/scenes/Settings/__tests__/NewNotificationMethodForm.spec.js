import React from 'react';
import { NewNotificationMethodForm } from '../NewNotificationMethodForm';


it('renders correctly', () => {
  expect(shallow(<NewNotificationMethodForm createNotificationMethod={() => null} />)).toMatchSnapshot();
});
