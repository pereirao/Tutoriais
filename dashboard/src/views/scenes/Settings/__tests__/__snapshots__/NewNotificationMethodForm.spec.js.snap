// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MBForm
  className="newForm"
  field_spacer_width={16}
  onSubmit={[Function]}
  submit_from_last={true}
  validateSubmittable={[Function]}
>
  <NotificationTypeField
    name="notification_type"
  />
  <div
    className="fieldLabel"
  >
    Value
  </div>
  <NameField
    is_required={true}
    name="value"
    placeholder="e.g. 2123456789 or email@test.com"
  />
  <div
    className="fieldLabel"
  >
    Label
  </div>
  <NameField
    is_required={false}
    name="label"
    placeholder="e.g. Brian's (Manager) Phone"
  />
  <MBFormSubmitButton
    inverted={false}
    is_loading={false}
    text="Submit"
    title="Submit form"
  />
  <NewFormError />
</MBForm>
`;
